src/detect-filestore.c

```diff
@@ -264,6 +264,15 @@ static int DetectFilestoreMatch (ThreadVars *t, DetectEngineThreadCtx *det_ctx,
      * matches. */
     if (file != NULL) {
         file_id = file->file_store_id;
+
+        if (s->id) {
+            if (file->sid_sze > file->sid_cap) {
+                file->sid = SCRealloc(file->sid, sizeof(uint32_t) * file->sid_cap * 2);
```

Comment:
2 problems here:
1. result not checked
2. memory leak on realloc failure.

Please see other examples of SCRealloc in the code for how to do the error checking properly.